web/js/jquery.jmri.js

```diff
@@ -965,7 +965,7 @@
                             if (h) {
                                 h.call(this, m);
                             } else if (!m.type) {
-                                jmri.log("ERROR: missing type property in " + m);
+                                console.error("ERROR: missing type property in " + m);
```

Comment:
Note: You *cannot* rely on the console existing--there is no reason not to expect `console.error` to crash out the JavaScript completely on user's devices, so please revert this, or come up with a reliable alternative.

There is *zero* benefit to writing code to making unit testing easy if if crashes out entirely for users.

Reply:
I see this as an aid to developer debugging.  Sending everything to console.log means we aren't using the developer tools provided to the browser effectively, because we aren't using the tools the browser gives us to flag error messages.  Per https://www.w3schools.com/jsref/met_console_error.asp every modern browser supports console.error, so we should be using that.

If we have to check to see if the console exists and that each function is available, I'll make that happen, but it will be wrapped inside of a new logger object that handles the checks appropriately.